src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -237,8 +234,76 @@ private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
                 queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type, _exportJobRecord.ResourceType));
             }
 
+            var filteredResources = new HashSet<string>();
+            if (_exportJobRecord.Filters != null)
+            {
+                foreach (var filter in _exportJobRecord.Filters)
+                {
+                    filteredResources.Add(filter.Type);
+                }
+            }
+
             IAnonymizer anonymizer = IsAnonymizedExportJob() ? await CreateAnonymizerAsync(cancellationToken) : null;
 
+            if (!progress.FinishedFilters)
+            {
+                var index = -1;
+                if (_exportJobRecord.Filters != null)
+                {
+                    foreach (var filter in _exportJobRecord.Filters)
+                    {
+                        index++;
+
+                        if (progress.Filter != null && filter != progress.Filter)
+                        {
+                            continue;
+                        }
+
+                        if (_exportJobRecord.ExportType == ExportJobType.All || filter.Type.Equals(KnownResourceTypes.Patient, StringComparison.Ordinal))
+                        {
+                            progress.SetFilter(filter);
+
+                            List<Tuple<string, string>> filterQueryParametersList = new List<Tuple<string, string>>(queryParametersList);
+                            foreach (var param in filter.Parameters)
+                            {
+                                filterQueryParametersList.Add(param);
+                            }
+
+                            await SearchWithFilter(exportJobConfiguration, progress, filter.Type, filterQueryParametersList, sharedQueryParametersList, anonymizer, "filter" + index + "-", cancellationToken);
+
+                            progress.SetFilter(null);
+                        }
+                    }
+                }
+
+                progress.MarkFiltersFinished();
+                await UpdateJobRecordAsync(cancellationToken);
+            }
+
+            if (_exportJobRecord.ExportType == ExportJobType.All || !filteredResources.Contains(KnownResourceTypes.Patient))
```

Comment:
>Contains [](start = 87, length = 8)

nit: shouldn't this use  StringComparison.Ordinal as in loop above?

Reply:
I don't know if the HashSet can take a StringComparison for its contains method. I'll look into it.